uncertainty_baselines/models/segmenter_be.py

```diff
@@ -0,0 +1,211 @@
+      ))
+
+    if self.backbone_configs.type == 'vit':
+      x, out = segmenter.ViTBackbone(
+          mlp_dim=self.backbone_configs.mlp_dim,
+          num_layers=self.backbone_configs.num_layers,
+          num_heads=self.backbone_configs.num_heads,
+          patches=self.patches,
+          hidden_size=self.backbone_configs.hidden_size,
+          dropout_rate=self.backbone_configs.dropout_rate,
+          attention_dropout_rate=self.backbone_configs.attention_dropout_rate,
+          classifier=self.backbone_configs.classifier,
+          name='backbone')(
+              x, train=train)
+    elif self.backbone_configs.type == 'vit_be':
+      # import pdb; pdb.set_trace()
+      x, out = ViTBackboneBE(
+          mlp_dim=self.backbone_configs.mlp_dim,
+          num_layers=self.backbone_configs.num_layers,
+          num_heads=self.backbone_configs.num_heads,
+          patches=self.patches,
+          hidden_size=self.backbone_configs.hidden_size,
+          dropout_rate=self.backbone_configs.dropout_rate,
+          attention_dropout_rate=self.backbone_configs.attention_dropout_rate,
+          classifier=self.backbone_configs.classifier,
+          ens_size=self.backbone_configs.ens_size,
+          random_sign_init=self.backbone_configs.random_sign_init,
+          be_layers=self.backbone_configs.be_layers,
+          name='backbone')(
+              x, train=train)
+    else:
+      raise ValueError(f'Unknown backbone: {self.backbone_configs.type}.')
+
+    if self.decoder_configs.type == 'linear':
+      output_projection = nn.Dense(
+        self.num_classes,
+        kernel_init=nn.initializers.zeros,
+        name='output_projection')
+    elif self.decoder_configs.type == 'linear_be':
+      output_projection = ed.nn.DenseBatchEnsemble(
+          self.num_classes,
+          self.backbone_configs.ens_size,
+          activation=None,
+          alpha_init=ed.nn.utils.make_sign_initializer(
+              self.backbone_configs.get("random_sign_init")),
+          gamma_init=ed.nn.utils.make_sign_initializer(
+              self.backbone_configs.get("random_sign_init")),
+          kernel_init=self.head_kernel_init,
+          name="output_projection_be")
+    else:
+      raise ValueError(
+          f'Decoder type {self.decoder_configs.type} is not defined.')
+
+    ens_size = self.backbone_configs.get("ens_size")
+
+    # Linear head only, like Segmenter baseline:
+    # https://arxiv.org/abs/2105.05633
+    x = jnp.reshape(x, [b*ens_size, gh, gw, -1])
+    x = output_projection(x)
+
+    # Resize bilinearly:
+    x = jax.image.resize(x, [b*ens_size, h, w, x.shape[-1]], 'linear')
+    out['logits'] = x
+
+    new_input_shape = tuple([input_shape[0]*ens_size,] + list(input_shape[1:-1]))
+    assert new_input_shape == x.shape[:-1], (
+        'BE Input and output shapes do not match: %d vs. %d.', new_input_shape,
+        x.shape[:-1])
+
+    return x, out
+
+
+def segmenter_be_transformer(num_classes: int,
```

Comment:
We've actually been moving to using just the class implementations, without the need for the functional alias. How about importing SegVitBE in `__init__.py`? You could call it, say, `SegmenterBE`. Similar for `segmenter.py`.

Reply:
I changed this but I was wondering why we used functional aliases before and not now? thank you!

Reply:
The original motivation was to have consistency where some models are Python functions that build Keras models through their functional API (ex: https://github.com/google/uncertainty-baselines/blob/main/uncertainty_baselines/models/wide_resnet.py#L144). I think that consistency resulted in more confusion in clarity though.